snuba/state/__init__.py

```diff
@@ -214,7 +217,7 @@ def record_query(query_metadata: Mapping[str, Any]) -> None:
         ).execute()
 
         if kfk is None:
-            kfk = Producer({"bootstrap.servers": ",".join(settings.DEFAULT_BROKERS)})
+            kfk = Producer(build_default_kafka_producer_configuration())
```

Comment:
There is a bug here, this does not behave as before. By using `build_default_kafka_producer_configuration` you are calling into `build_kafka_producer_configuration` which applies these two parameters which are not default:
```
"partitioner": "consistent",
"message.max.bytes": 50000000,  # 50MB, default is 1MB
```

You may want to remove those two default parameters from `build_kafka_producer_configuration` and always rely on overrides to set them when needed since they are different from the librdkafka defaults:
https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md

Reply:
Fair enough. Updated to remove default parameters.